src/coverlet.core/Symbols/CecilSymbolHelper.cs

```diff
@@ -714,6 +729,43 @@ static Instruction GetPreviousNoNopInstruction(Instruction i)
             }
         }
 
+        private bool SkipExpressionBreakpointsSequences(MethodDefinition methodDefinition, Instruction instruction)
+        {
+            if (_sequencePointOffsetToSkip.ContainsKey(methodDefinition.FullName) && _sequencePointOffsetToSkip[methodDefinition.FullName].Contains(instruction.Offset) && instruction.OpCode == OpCodes.Nop)
+            {
+                return true;
+            }
+            /* 
+               Sequence to skip https://github.com/dotnet/roslyn/blob/master/docs/compilers/CSharp/Expression%20Breakpoints.md
+               // if (1 == 0)
+               // sequence point: (line 33, col 9) to (line 40, col 10) in C:\git\coverletfork\test\coverlet.core.tests\Samples\Instrumentation.SelectionStatements.cs
+               IL_0000: ldc.i4.1
+               IL_0001: brtrue.s IL_0004
+               // if (value is int)
+               // sequence point: (line 34, col 9) to (line 40, col 10) in C:\git\coverletfork\test\coverlet.core.tests\Samples\Instrumentation.SelectionStatements.cs
+               IL_0003: nop
+                // sequence point: hidden
+                ...
+             */
+            if (
+                    instruction.OpCode == OpCodes.Ldc_I4 && instruction.Operand is int operandValue && operandValue == 1 &&
+                    instruction.Next?.OpCode == OpCodes.Brtrue &&
```

Comment:
To check exactly above pattern you should also check that target of `brtrue.s` is instruction after `nop`

Reply:
Uh you're right thanks.

Reply:
According to the pattern I also checked ```methodDefinition.DebugInformation.GetSequencePoint(instruction.Next?.Next?.Next).IsHidden```. That's also what I observed in the IL that after the ```nop``` instruction a hidden sequence point follows. But I'm not a specialist here and maybe this isn't important.